virtme/commands/run.py

```diff
@@ -1842,6 +1880,28 @@ def get_net_mac(index):
     # Load a normal kernel
     qemuargs.extend(["-kernel", kernel.kimg])
     if kernelargs:
+        if args.systemd:
+            init_environment_vars = []
+            for arg in kernelargs:
+                match = re.match(r"(virtme_.*)=(.*)", arg)
+                if not match:
+                    continue
+                init_environment_vars.append(f"{match.group(1)}={match.group(2)}")
+            os.makedirs(CACHE_DIR, exist_ok=True)
+            with open(SERIAL_GETTY_FILE, "w", encoding="utf-8") as f:
+                f.write(
+                    f"""[Service]
+StandardInput=tty
+StandardOutput=tty
+TTYPath=/dev/%I
+TTYReset=yes
+TTYVHangup=yes
+Environment={shlex.join(init_environment_vars)}"""
+                )
+                if args.root == "/":
+                    f.write(f"\nExecStart={guest_tools_path}/{virtme_init_cmd}")
+                else:
+                    f.write(f"\nExecStart=/run/virtme/guesttools/{virtme_init_cmd}")
```

Comment:
What exactly is the purpose of still spawning the vng inits?
Also it might be cleaner to use `systemd-run-generator` though if tty connection is desired I think we would still need at least a drop-in

Reply:
Oh I just saw this comment after pushing the new changes... Thanks for the pointers. It's a good approach but then we would only support roots with systemd > 255, right?